### PR TITLE
Add rate limiter caching to gem

### PR DESCRIPTION
**Pull request by yashshah1**

### CODE REVIEW

File: lib/rsalesloft/connection.rb [hunk 1]
```diff
@@ -19,30 +18,54 @@ def post(path, req_body)
...
+          # key once the minute changes.
+          conn.set(redis_key_for_remaining_credits, headers["x-ratelimit-remaining-minute"], ex: 65)
+  
+          # The max credits are more stable, only refresh once every day
+          conn.set(RSalesloft::Config.redis_key_for_max_credits, headers["x-ratelimit-limit-minute"] || 600, ex: 1.day.to_i)
+        end
+      end
+
+      def max_credits
+        redis_pool.with { |conn| (conn.get(RSalesloft::Config.redis_key_for_max_credits) || 600).to_i }
```
Comment (guimon) [inline lib/rsalesloft/connection.rb]: 600 is hardcoded

File: lib/rsalesloft/connection.rb [hunk 2]
```diff
@@ -19,30 +18,54 @@ def post(path, req_body)
...
       private
 
+      def update_remaining_credits(headers)
+        redis_pool.with do |conn|
+          # Since there is a new key every minute, expiring a little later is okay since we won't hit this
+          # key once the minute changes.
+          conn.set(redis_key_for_remaining_credits, headers["x-ratelimit-remaining-minute"], ex: 65)
+  
+          # The max credits are more stable, only refresh once every day
+          conn.set(RSalesloft::Config.redis_key_for_max_credits, headers["x-ratelimit-limit-minute"] || 600, ex: 1.day.to_i)
```
Comment (guimon) [inline lib/rsalesloft/connection.rb]: 600

File: lib/rsalesloft/connection.rb [hunk 3]
```diff
@@ -19,30 +18,54 @@ def post(path, req_body)
...
+        redis_pool.with { |conn| (conn.get(redis_key_for_remaining_credits) || max_credits).to_i }
       end
 
       private
 
+      def update_remaining_credits(headers)
+        redis_pool.with do |conn|
+          # Since there is a new key every minute, expiring a little later is okay since we won't hit this
+          # key once the minute changes.
+          conn.set(redis_key_for_remaining_credits, headers["x-ratelimit-remaining-minute"], ex: 65)
```
Comment (guimon) [inline lib/rsalesloft/connection.rb:49]: the key here should include the current minute, no?

Comment (guimon) [reply lib/rsalesloft/connection.rb:49]: nevermind I see it now

**Review by guimon — APPROVED** [review]
Nice
Just replace the 600s

File: lib/rsalesloft/connection.rb (hunk 1 above)
Comment (yashshah1) [reply lib/rsalesloft/connection.rb]: changed to config

**Pull request merged**